includes/components/class-user.php

```diff
@@ -55,18 +55,45 @@ public function __construct( $args = [] ) {
 
 			add_action( 'personal_options_update', [ $this, 'update_profile_fields' ], 100 );
 			add_action( 'edit_user_profile_update', [ $this, 'update_profile_fields' ], 100 );
+
+			// Set user status.
+			add_action( 'admin_init', [ $this, 'set_user_status' ] );
 		} else {
 
 			// Redirect author page.
 			add_action( 'template_redirect', [ $this, 'redirect_author_page' ] );
 
 			// Alter templates.
 			add_filter( 'hivepress/v1/templates/site_footer_block', [ $this, 'alter_site_footer_block' ] );
+
+			// Set user status.
+			add_action( 'init', [ $this, 'set_user_status' ] );
 		}
 
 		parent::__construct( $args );
 	}
 
+	/**
+	 * Updates user online status.
+	 */
+	public function set_user_status() {
+
+		// Get online users.
+		$online_users = (array) get_transient( 'online_users' );
+
+		// Get user ID.
+		$user_id = get_current_user_id();
+
+		if ( in_array( $user_id, array_keys( $online_users ) ) && $online_users[ $user_id ] > time() - 60 ) {
+            return;
+		}
+
+        // Add user.
+        $online_users[ $user_id ] = time();
+
+        set_transient( 'online_users', $online_users, 60 );
```

Comment:
The expiration is too low, especially if the website has no object caching it will create another DB record every minute (transients are stored in wp_options table).

The user activity time seems to be already stored as an array with IDs as keys and seconds as values, why the global transient expiration is set this low? It's possible to just clean it up (e.g. remove user IDs with too old timestamps from the array during another check) instead of refreshing the whole transient every minute